src/diffusers/models/unet_2d_blocks.py

```diff
@@ -1478,16 +1478,33 @@ def forward(
         cross_attention_kwargs = cross_attention_kwargs if cross_attention_kwargs is not None else {}
 
         for resnet, attn in zip(self.resnets, self.attentions):
-            # resnet
```

Comment:
Adding gradient checkpointing for the other two IF unet blocks

Reply:
Note that this means we were still doing gradient checkpointing previously for IF but just making less checkpoints

Reply:
For curiosity: in the earlier case which blocks were not being chrckpointed?

Reply:
SimpleCrossAttn{Down,Up}Block2D :)